tools/fast/docs/SHELL.md

```diff
@@ -0,0 +1,81 @@
+### The FAST Shell
+
+FAST has a [`Shell`](https://godoc.org/github.com/filecoin-project/go-filecoin/tools/fast#Filecoin.Shell) which can be used to start the users shell with the environment setup to run `go-filecoin` commands against the process it is called on.
+
+The exact environment the shell will have is largely dependent on the plugin use are using.
+Generally, this will be the Filecoin localplugin when writing tests using FAST.
+
+The Filecoin localplugin shell environment will have the following variables set in the shell for the user.
+
+| Name  | Description |
+|:---|:---|
+| `FIL_PATH` |  The value is set to the repository directory for the Filecoin node. Any `go-filecoin` commands ran in the shell will be executed against the Filecoin process for which the Shell method was invoked on in FAST. |
+| `FIL_PID` | The value is set to the PID for the Filecoin daemon. |
+| `FIL_BINARY` | The value is set to the binary running the Filecoin daemon. Please refer to the section below on `PATH` for more details. |
+| `PATH` | The users `PATH` will be updated to include a location that contains the binary used for executing all `go-filecoin` commands. The `go-filecoin` binary included in this location itself is defined by either the value of the `localplugin.AttrFilecoinBinary`, or the first `go-filecoin` binary found in the users `PATH`. <br/> <br/>_Note: The value of `FIL_BINARY` will not be the exact value. During node setup, the binary is copied to ensure it does not change during execution. `FIL_BINARY` will be this new path._ |
+
+The information around the `PATH` seems a little complex, but it's to ensure that there are no issues as a result of mixing binaries.
+This has the advantage that while using FAST, users can re-compile `go-filecoin`.
+It should be noted that the copying of the binary occurs during the call to `NewProcess`.
+
+It should also be noted that users shell configuration will be ran when the shell opens.
+If shell configuration updates the `PATH` by appending to the front, if any of those directories contain `go-filecoin`, then the `go-filecoin` command inside of the FAST Shell will **not** point to the correct binary.
```

Comment:
This seems like a trap that many people will hit, in particular because `go install` installs a binary to a path that many people will have intentionally prepended.

Is it possible to add some kind of check inside the shell if this has happened? Alternatively, update all docs to use `$FIL_BINARY` rather than `go-filecoin` to avoid path ambiguity?

FWIW I never `go install`, preferring to use `./go-filecoin` to be sure. Is this a reason to remove suggestions to explicitly install in this way, e.g. from the wiki?

Reply:
> Is it possible to add some kind of check inside the shell if this has happened?

A check could be added in the plugin itself to determine if `go-filecoin` already exists in the `PATH`, similarly to the check for `FIL_PATH`. Referencing `$FIL_PATH` might be the best option here.